test/Sentry.Tests/Internals/FileSystemTests.cs

```diff
@@ -0,0 +1,56 @@
+using System.IO.Abstractions.TestingHelpers;
+
+namespace Sentry.Tests.Internals;
+
+public class FileSystemTests
+{
+    private const string Filename = "lock.txt";
+
+    [Fact]
+    public void Leased_file_cannot_be_leased_again()
+    {
+        const string path = @$"C:\test\{Filename}";
+
+        IDictionary<string, MockFileData> files  = new Dictionary<string, MockFileData>
+        {
+            { path, new MockFileData(Filename) { AllowedFileShare = FileShare.None } }
+        };
+
+        IFileSystem fileSystem = new FakeFileSystem(files);
+
+        Assert.Throws<IOException>(() =>
+        {
+            using (fileSystem.GetLeaseFile(path)) { }
+        });
+    }
+
+    [Fact]
+    public void RealFileSystem_Leased_file_is_accessible_after_process_crush()
+    {
+        FileSystem.Instance.DeleteFile(Filename);
```

Comment:
We're going to have this test run concurrently once per each TFM in the test project.
Ideally we use a `guid` as a file name here and make sure we delete the file in a `finally` block in the test to avoid leaving trash behind